app/main/views/suppliers.py

```diff
@@ -85,13 +86,84 @@ def supplier_details():
     except APIError as e:
         abort(e.status_code)
     supplier['contact'] = supplier['contactInformation'][0]
+    country_name = get_country_name_from_country_code(supplier['registrationCountry'])
 
     return render_template(
         "suppliers/details.html",
         supplier=supplier,
+        country_name=country_name,
     ), 200
 
 
+@main.route('/registered-address/edit', methods=['GET'])
+@login_required
+def edit_registered_address(registered_address_form=None, registered_country_form=None, error=None):
```

Comment:
Yeah, I spent a bit of time at the end of yesterday getting this working and it ends up being really quite messy. Rather than populating the data in the forms when it's instantiated by passing in the data as kwargs with **supplier (or whatever), you need to assign each one individually.
I think it makes sense on views with forms without many fields, but once you have a few it gets a bit cumbersome.

Reply:
OK, i did the thing.